### ISSUE TITLE
Runc runtime using same image as in kata creates mount failure

### ISSUE THREAD

**Issue by David9902** [confidential-containers/guest-components#137]
Hi everyone, 
I am not 100% sure if this is the right place for the issue. Feel free to move it if needed. 

I installed additionally to ConfidentialContainers the runc ruapiVersion: node.k8s.io/v1
```
apiVersion: node.k8s.io/v1
kind: RuntimeClass
metadata:
  name: runc-rc
handler: runc
```

when deploying 2 pods using the same image, one with kata-clh runtime from confidential-container project and one pod with runc runtime I am running into following issue on the kata-clh pod side: 

```
[root@rombach-dev timing]# kc describe pod nginx
Name:                nginx
Namespace:           default
Priority:            0
Runtime Class Name:  kata-clh
Service Account:     default
Node:                rombach-dev.novalocal/192.168.0.66
Start Time:          Tue, 11 Apr 2023 10:51:31 +0300
Labels:              run=nginx
Annotations:         cni.projectcalico.org/containerID: f057a1729d668dffd779e0e5cbdad3f5fe210090537727cef55d0181a9adcf42
                     cni.projectcalico.org/podIP: 192.168.247.73/32
                     cni.projectcalico.org/podIPs: 192.168.247.73/32
Status:              Running
IP:                  192.168.247.73
IPs:
  IP:  192.168.247.73
Containers:
  nginx:
    Container ID:   containerd://b61dfbc4e55940486bc0ec94954e2d19d2bc1a7d7d0db10d736d56880bd5e98f
    Image:          192.168.0.66:443/nginx-1:1.22.0
    Image ID:       [Encrypted]
    Port:           <none>
    Host Port:      <none>
    State:          Waiting
      Reason:       RunContainerError
    Last State:     Terminated
      Reason:       StartError
      Message:      failed to create containerd task: failed to create shim task: failed to mount "/run/kata-containers/shared/containers/b61dfbc4e55940486bc0ec94954e2d19d2bc1a7d7d0db10d736d56880bd5e98f/rootfs" to "/run/kata-containers/b61dfbc4e55940486bc0ec94954e2d19d2bc1a7d7d0db10d736d56880bd5e98f/rootfs", with error: ENOENT: No such file or directory: unknown
      Exit Code:    128
      Started:      Thu, 01 Jan 1970 02:00:00 +0200
      Finished:     Tue, 11 Apr 2023 10:51:35 +0300
    Ready:          False
    Restart Count:  1
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-zv4rf (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  kube-api-access-zv4rf:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              node-role.kubernetes.io/worker=
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age              From               Message
  ----     ------     ----             ----               -------
  Normal   Scheduled  5s               default-scheduler  Successfully assigned default/nginx to rombach-dev.novalocal
  Normal   Pulled     2s (x2 over 2s)  kubelet            Container image "192.168.0.66:443/nginx-1:1.22.0" already present on machine
  Normal   Created    2s (x2 over 2s)  kubelet            Created container nginx
  Warning  Failed     2s (x2 over 2s)  kubelet            Error: failed to create containerd task: failed to create shim task: failed to mount "/run/kata-containers/shared/containers/nginx/rootfs" to "/run/kata-containers/nginx/rootfs", with error: ENOENT: No such file or directory: unknown
  Warning  BackOff    1s               kubelet            Back-off restarting failed container nginx in pod nginx_default(80f6a667-b12a-4478-b7e2-e255b710a25d)
```

The pods are created as follows: 
```
apiVersion: v1
kind: Pod
metadata:
  labels:
    run: nginx
  name: nginx
spec:
  containers:
  - image: 192.168.0.66:443/nginx-1:1.22.0
    name: nginx
  dnsPolicy: ClusterFirst
  runtimeClassName: kata-clh
```

```
apiVersion: v1
kind: Pod
metadata:
  labels:
    run: nginx
  name: nginx-runc
spec:
  containers:
  - image: 192.168.0.66:443/nginx-1:1.22.0
    name: nginx
  dnsPolicy: ClusterFirst
  runtimeClassName: runc-rc
```

The problem is solved when using different image tags, so kata-clh does not run into this line of finding the image: 
 ` Normal   Pulled     2s (x2 over 2s)  kubelet            Container image "192.168.0.66:443/nginx-1:1.22.0" already present on machine
`

**Comment by arronwy** [comment]
Can you try with this PR: https://github.com/kata-containers/kata-containers/pull/6457 to rebuild kata-agent, which try to fix the issue pod  with multiple same images .

**Comment by arronwy** [comment]
BTW, please set `imagePullPolicy: Always` in pod yaml when use CoCo to avoid miss use the host images.

**Comment by David9902** [comment]
> Can you try with this PR: [kata-containers/kata-containers#6457](https://github.com/kata-containers/kata-containers/pull/6457) to rebuild kata-agent, which try to fix the issue pod with multiple same images .

Sorry I am currently a bit short on time, so my main purpose was to make devs aware of this issue. But good that there seems to be a fix already available. 

Personally I would just work for now with 2 different image names, until it is fixed. (Will it be part of v0.5 ?)

**Comment by arronwy** [comment]
> > Can you try with this PR: [kata-containers/kata-containers#6457](https://github.com/kata-containers/kata-containers/pull/6457) to rebuild kata-agent, which try to fix the issue pod with multiple same images .
> 
> Sorry I am currently a bit short on time, so my main purpose was to make devs aware of this issue. But good that there seems to be a fix already available.
> 
> Personally I would just work for now with 2 different image names, until it is fixed. (Will it be part of v0.5 ?)

Yes, this fix will be part of v0.5

**Comment by Xynnn007** [comment]
@David9902 @arronwy Can this issue be closed?
